contracts/voting-verifier/src/state.rs

```diff
@@ -69,13 +43,29 @@ impl Poll {
     }
 }
 
+#[cw_serde]
+pub struct PollMessage {
+    pub msg: Message,
+    pub poll_id: PollID,
+    pub index_in_poll: usize,
+}
+
+impl PollMessage {
+    pub fn new(msg: Message, poll_id: PollID, index_in_poll: usize) -> Self {
+        Self {
+            msg,
+            poll_id,
+            index_in_poll,
+        }
+    }
+}
+
 pub const POLL_ID: counter::Counter<PollID> = counter::Counter::new("poll_id");
 
 pub const POLLS: Map<PollID, Poll> = Map::new("polls");
 
-pub const PENDING_MESSAGES: Map<PollID, Vec<Message>> = Map::new("pending_messages");
-
-pub const VERIFIED_MESSAGES: Map<&CrossChainId, Message> = Map::new("verified_messages");
+// key is message `hash_id`
+pub const POLL_MESSAGES: Map<&[u8; 32], PollMessage> = Map::new("poll_messages");
```

Comment:
I feel like it might be cleaner to use a named type for the key.

Reply:
yes, please alias the key